python/tvm/script/builder/tir/block_frame.py

```diff
@@ -44,6 +45,7 @@ def init() -> BlockInitFrame:
 
 
 def where(predicate) -> None:
+    predicate = cast(predicate, "bool")
```

Comment:
in principle, we should implement the logic as much as we could on c++ side, so that the C++/python behavior are consistent. in our case, shall we move the casting to c++?

https://github.com/cyx-6/tvm/blob/8b5e6d2f819d68b13dd6a7be0d822ec94d2b894c/src/script/builder/tir/block_frame.cc#L101-L108